### PR TITLE
Add 'How to deploy' doc

### PR DESCRIPTION
**Pull request by TBonnin**
## What does this change?

Add 'How to deploy' documentation steps
## What is the value of this and can you measure success?

New comers have a reference doc when learning how to deploy Frontend to production
## Does this affect other platforms - Amp, Apps, etc?

No
## Request for comment

@gtrufitt @johnduffell @NataliaLKB @jfsoul @SiAdcock @sndrs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ Welcome! To get set up, please follow [the installation guide](docs/01-start-her
 
 [The best place to start is here](docs/01-start-here)
 
-Fixes for common problems can be found [here](docs/01-start-here/03-troubleshooting.md).
+Fixes for common problems can be found [here](docs/01-start-here/04-troubleshooting.md).
```
Comment (TBonnin) [inline README.md]: That's kind of the problem with ordering being part of the name. Any manually created link to a file must be updated if new file is added or order changes.
Let's see if it becomes a major problem.
Eventually I would like to revamp the main README as well

File: docs/01-start-here/03-how-to-deploy.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+
+##Blocking deployment
+
+In order to prevent anybody to deploy you can lock deployment. To do so, use the `goo` tool:
+`./goo deploy block`
+
+Use `./goo deploy unblock` to unblock deployment
+
+##Frequent questions
```
Comment (jfsoul) [inline docs/01-start-here/03-how-to-deploy.md]: Being pedantic I would call this "Frequently asked questions" and bullet point them.

File: docs/01-start-here/03-how-to-deploy.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# How to deploy
+
+Your pull request is ready, all comments made by reviewers have been addressed and CI results (Teamcity) are green.
+Congratulations, you can deploy your code to production. Here are the steps to do so:
+
+##Deploying
+
+1. Merge your pull request to master
+2. Wait for the tests and the build to finish (Teamcity)
```
Comment (gtrufitt) [inline docs/01-start-here/03-how-to-deploy.md]: Link to TC?

File: docs/01-start-here/03-how-to-deploy.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+##Frequent questions
+
+Where is the `goo` tool? 
+`goo` is available in The Guardian platform [repo](https://github.com/guardian/platform/) (Private)
+
+When something goes wrong during deployment, a service cluster might be left in an inconsistent state (ie: running servers count doesn't match the expected count)
+Developers would need to fix this inconsistency before to start any subsequent deploy.
+To do so, you can modify the expected number of instance for a given Autoscaling group using the AWS console or `goo` tool (ex: `./goo groups update AUTOSCALING-GROUP-ID 3 3 12`)
+
```
Comment (jfsoul) [inline docs/01-start-here/03-how-to-deploy.md]: Good call including something about inconsistent state.  You could reference this in the deploy steps as well potentially as it is relevant there.

File: docs/01-start-here/03-how-to-deploy.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+Your pull request is ready, all comments made by reviewers have been addressed and CI results (Teamcity) are green.
+Congratulations, you can deploy your code to production. Here are the steps to do so:
+
+##Deploying
+
+1. Merge your pull request to master
+2. Wait for the tests and the build to finish (Teamcity)
+3. Once the build is done, it would be automatically deployed to CODE (The Guardian staging environment)
+4. Verify that your change works properly on CODE
+5. You can find out who is also currently deploying by using the [Deploy Radiator](https://frontend.gutools.co.uk/deploys-radiator)
```
Comment (gtrufitt) [inline docs/01-start-here/03-how-to-deploy.md]: Or in #dotcom-push slack room

**Comment by jfsoul** [comment]
This is great! I can tell you it definitely would have been useful to me 👍

File: docs/01-start-here/03-how-to-deploy.md (hunk 5 above)
Comment (gtrufitt) [inline docs/01-start-here/03-how-to-deploy.md]: Oh you mention that below 💃

File: docs/01-start-here/03-how-to-deploy.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+6. If several people are pushing to PROD at the same time, please coordinates with each other. Slack channel #dotcom-push is a good place to do so. 
+7. When ready use `goo` tool to deploy the build to production. (ex: `./goo deploy --prod -b BUILD_NUMBER` and then confirm). You will need valid AWS credentials. Please use [Janus](https://janus.gutools.co.uk/#) to obtain them.
```
Comment (gtrufitt) [inline docs/01-start-here/03-how-to-deploy.md]: We can deploy the latest without the `-b BUILD_NUMBER` but do we want to teach people to always put build number?

Comment (gtrufitt) [inline docs/01-start-here/03-how-to-deploy.md]: Where do you get the build number? Screenshots might be useful?

image

image

Comment (gtrufitt) [inline docs/01-start-here/03-how-to-deploy.md]: Link to the goo tool readme?)

File: docs/01-start-here/03-how-to-deploy.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
+# How to deploy
+
+Your pull request is ready, all comments made by reviewers have been addressed and CI results (Teamcity) are green.
+Congratulations, you can deploy your code to production. Here are the steps to do so:
+
+##Deploying
+
+1. Merge your pull request to master
+2. Wait for the tests and the build to finish (Teamcity)
+3. Once the build is done, it would be automatically deployed to CODE (The Guardian staging environment)
```
Comment (jfsoul) [inline docs/01-start-here/03-how-to-deploy.md]: ~~would~~ should

File: docs/01-start-here/03-how-to-deploy.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+`./goo deploy block`
+
+Use `./goo deploy unblock` to unblock deployment
+
+##Frequent questions
+
+Where is the `goo` tool? 
+`goo` is available in The Guardian platform [repo](https://github.com/guardian/platform/) (Private)
+
+When something goes wrong during deployment, a service cluster might be left in an inconsistent state (ie: running servers count doesn't match the expected count)
```
Comment (gtrufitt) [inline docs/01-start-here/03-how-to-deploy.md]: "Which may cause a failed status in RiffRaff"

image

**Comment by gtrufitt** [comment]
Awesome! This is super useful 👍

File: docs/01-start-here/03-how-to-deploy.md (hunk 3 above)
Comment (TBonnin) [inline docs/01-start-here/03-how-to-deploy.md]: Yep. I will add

File: docs/01-start-here/03-how-to-deploy.md (hunk 2 above)
Comment (TBonnin) [inline docs/01-start-here/03-how-to-deploy.md]: The non-native english speaker should not be in charge of the doc 😜

File: docs/01-start-here/03-how-to-deploy.md (hunk 6 above)
Comment (TBonnin) [inline docs/01-start-here/03-how-to-deploy.md]: I prefer to be explicit and teach people to enter the build number.

Comment (gtrufitt) [inline docs/01-start-here/03-how-to-deploy.md]: 👌

**Pull request merged**